src/deepforest/evaluate.py

```diff
@@ -206,32 +213,39 @@ def evaluate_boxes(predictions, ground_df, iou_threshold=0.4):
         ):
             # Create geometry from bounding box columns
             ground_df = ground_df.copy()
-            ground_df["geometry"] = ground_df.apply(
-                lambda x: shapely.geometry.box(x.xmin, x.ymin, x.xmax, x.ymax), axis=1
+            ground_df["geometry"] = shapely.box(
+                ground_df["xmin"], ground_df["ymin"], ground_df["xmax"], ground_df["ymax"]
             )
         ground_df = gpd.GeoDataFrame(ground_df, geometry="geometry")
 
+    # Pre-group predictions by image
+    predictions_by_image = {
+        name: group.reset_index(drop=True)
+        for name, group in predictions.groupby("image_path")
+    }
+
     # Run evaluation on all plots
     results = []
     box_recalls = []
     box_precisions = []
     for image_path, group in ground_df.groupby("image_path"):
-        # clean indices
-        image_predictions = predictions[
-            predictions["image_path"] == image_path
-        ].reset_index(drop=True)
+        # Predictions for this image
+        image_predictions = predictions_by_image.get(image_path, pd.DataFrame())
+        if not isinstance(image_predictions, pd.DataFrame) or image_predictions.empty:
+            image_predictions = pd.DataFrame()
 
         # If empty, add to list without computing IoU
         if image_predictions.empty:
             result = pd.DataFrame(
                 {
                     "truth_id": group.index.values,
-                    "prediction_id": None,
-                    "IoU": 0,
-                    "predicted_label": None,
-                    "score": None,
-                    "match": False,
-                    "true_label": group.label,
+                    "prediction_id": pd.Series([None] * len(group), dtype="object"),
```

Comment:
Just a note that the reason this is so verbose, is because it avoids a FutureWarning where pandas will complain about concatenating empty elements. This seems to fix the issue by giving everything explicit types so it doesn't complain about mixed columns.